xpreports/windows/managedinstalls.go

```diff
@@ -0,0 +1,150 @@
+package windows
+
+import (
+	"encoding/json"
+	"os/exec"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+// GetManagedInstalls is a crappy representation of things installed on a windows client
+func get32BitPrograms() ([]Program, error) {
+	cmd := exec.Command("powershell", "gp", "HKLM:\\Software\\Microsoft\\Windows\\CurrentVersion\\Uninstall\\*", "|", "ConvertTo-Json")
+
+	// cmd.Stderr = os.Stderr
+	o, err := cmd.Output()
+	if err != nil {
+		return nil, errors.Wrap(err, "exec gp 32 bit programs")
+	}
+
+	var p []Program
+
+	if err := json.Unmarshal(o, &p); err != nil {
```

Comment:
At the end of the functions you can make the returns shorter by just writing
```
err = json.Unmarshal....
return p, errors.Wrap(err, "reason...)
```

`errors.Wrap` will return nil if the underlying error is nil.